lightning-invoice/src/lib.rs

```diff
@@ -1415,6 +1415,15 @@ impl Deref for SignedRawInvoice {
 	}
 }
 
+impl core::hash::Hash for InvoiceSignature {
```

Comment:
Looks like clippy doesn't like this

```
error: you are implementing `Hash` explicitly but have derived `PartialEq`
    --> lightning-invoice/src/lib.rs:1418:1
     |
1418 | / impl core::hash::Hash for InvoiceSignature {
1419 | |     fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
1420 | |         let (recovery_id, sig) = self.0.serialize_compact();
1421 | |         let recovery_id = recovery_id.to_i32();
...    |
1424 | |     }
1425 | | }
     | |_^
     |
     = note: `#[deny(clippy::derive_hash_xor_eq)]` on by default
note: `PartialEq` implemented here
    --> lightning-invoice/src/lib.rs:456:28
     |
456  | #[derive(Clone, Debug, Eq, PartialEq)]
     |                            ^^^^^^^^^
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_hash_xor_eq
     = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Ok I'll leave it open, thank you

Reply:
I'd argue clippy isn't dumb here. The contract of `Hash` states that hashes have to be equal if `Eq` returns equal for the hashed objects. With this implementation we'd rely on libsecp256k1 to not use the higher bits of the recovery id byte for other things for that contract to hold. Related to https://github.com/rust-bitcoin/rust-secp256k1/issues/463. The `Hash` impl from `secp256k1` would be guaranteed to be at least internally consistent (just not between versions I guess).